include/rocksdb/file_system.h

```diff
@@ -132,6 +132,25 @@ struct IODebugContext {
   // To be set by the FileSystem implementation
   std::string msg;
 
+  // To be set by the underlying FileSystem implementation.
+  std::string request_id;
+
+  // In order to log required information in IO tracing for different
+  // operations, Each bit in trace_data stores which corresponding info from
+  // IODebugContext will be added in the trace. Foreg, if trace_data = 1, it
+  // means bit at position 0 is set so TraceOp::kRequestID (request_id) will be
+  // logged in the trace record.
+  //
+  // In order to log the required info, trace_data needs to be updated as
```

Comment:
Why not just log whatever's in request_id? If the FS implementation didn't set it, it'll be an empty string.

Reply:
Yes, I can add it like filename. It will be empty string if not set. I thought we don't want empty string in trace so I added the logic. I will make the request_id default.

Reply:
That's a  good point. What would the trace look like with and without empty string (I assume there will be some delimiters)? It might not be a bad thing to leave it there if it makes the trace easier to parse.

Reply:
It will be something like ```"RequestID:   , FileOperation: Append"```. Since its a string, I can add a check at the parser side as well to  add request id only if it's not empty, if that's fine.

Reply:
But in future, if WS want log additional information of type int then we might need to switch back to this approach.

Reply:
> It will be something like `"RequestID: , FileOperation: Append"`. Since its a string, I can add a check at the parser side as well to add request id only if it's not empty, if that's fine.

Yeah, either checking on the parser side, or providing a setter in ```IODebugContext``` might be better as it'll hide the implementation details from WS.